src/core/path.c

```diff
@@ -196,9 +195,30 @@ QuicConnGetPathForDatagram(
 
     if (Connection->PathsCount == QUIC_MAX_PATH_COUNT) {
         //
-        // Already tracking the maximum number of paths.
+        // See if any old paths share the same remote address, and is just a rebind.
+        // If so, remove the old paths.
+        // NB: Traversing the array backwards is simpler and more efficient here due
+        // to the array shifting that happens in QuicPathRemove.
         //
-        return NULL;
+        for (uint8_t i = Connection->PathsCount - 1; i > 0; i--) {
+            BOOLEAN IsSimilarRemoteAddress =
+                QuicAddrGetFamily(&Datagram->Tuple->RemoteAddress) == QuicAddrGetFamily(&Connection->Paths[i].RemoteAddress) &&
+                QuicAddrCompareIp(&Datagram->Tuple->RemoteAddress, &Connection->Paths[i].RemoteAddress);
+            BOOLEAN IsSameLocalAddress = QuicAddrCompare(&Datagram->Tuple->LocalAddress, &Connection->Paths[i].LocalAddress);
+            if (IsSimilarRemoteAddress
+                && IsSameLocalAddress
+                && !Connection->Paths[i].IsActive) {
```

Comment:
You can optimize this `if` without separate state variables and put the `IsActive` check first.